### DISCUSSION TITLE
Unable to to debug PHP using a template in github codespaces

### DISCUSSION THREAD

**Discussion by jtagaca** [community/community#35502] [Codespaces]
Hi, I am using this template https://.com/jimblizzard/codespaces-for-php-demo, but it doesn't seem to work when using  codespaces with xdebug in php.

Here's a video of my issue.
https://drive.google.com/file/d/1sePxxWWbtPSNLCk5oiqpnWsYPjkW4-GR

As you can see, my server is listening on port 8000, and my debugger server is listening on port 9000. The application, however, freezes and is unable to reach my debug points. I attempted to resolve this issue, but it appears that the solution is not available online, and there are no resources that could help. I tried several times to see if the problem was on my end, but everything appears to be normal on my configuration.

I already spent many many hours trying to solve this issue but I am very stuck. 

Any help would be appreciated!

Thank you!

**Comment by jfullstackdev** [comment]
> https://.com

that seems to be an error in URL ?

**Reply by jtagaca** [reply]
Hi thanks so much for responding, could you elaborate more? I don't think that would be the issue since these are remote servers that are using my local ports to run?

**Comment by legomushroom** [comment]
The issue might be that the ports are either bound to a different port, or the port-forwarding logic has a bug. Mind seeing the `ports` tab in VSCode (next to the terminal), what are the `local address` values for those ports?


<img width="1392" alt="image" src="https://user-images.githubusercontent.com/1478800/194960252-aa7771b1-2140-4d58-97b7-3c8348be26dc.png">

**Reply by jtagaca** [reply]
Hi, thanks for responding here are the local ports!

https://drive.google.com/file/d/1rWZ4F4LCKTxJo29VsUt_ApBjWR1McE4U

here is my launch.json 

```
{
    "version": "0.2.0",
    "configurations": [
        
        {
            "name": "Launch Built-in web server",
            "type": "php",
            "request": "launch",
            "runtimeArgs": [
                "-dxdebug.mode=debug",
                "-dxdebug.start_with_request=yes",
                "-S",
                "localhost:8000"
            ],
            "program": "",
            "cwd": "${workspaceRoot}",
            "port": 9003,
            "serverReadyAction": {
                "pattern": "Development Server \\(http://localhost:([0-9]+)\\) started",
                "uriFormat": "http://localhost:%s",
                "action": "openExternally"
            }
        },

      {
        "name": "Listen for Xdebug 2 (Legacy)",
        "type": "php",
        "request": "launch",
        "port": 9000
      },
        {
            "name": "Launch application",
            "type": "php",
            "request": "launch",
            "program": "${workspaceFolder}/index.php",
            "cwd": "${workspaceFolder}",
            "port": 9000
        },
        {
            "name": "Listen for XDebug",
            "type": "php",
            "request": "launch",
            "port": 9000
        },
        {
            "name": "Launch currently open script",
            "type": "php",
            "request": "launch",
            "program": "${file}",
            "cwd": "${fileDirname}",
            "port": 9000
        }
    ]
}
```

Let me know if you have any questions!

Thanks again!!

**Reply by legomushroom** [reply]
Hey @jtagaca 👋

The port number you specify in the config is the port inside your Codespace. We try to bind to the same port number on your local machine, but if, for some reason, we cannot, a random port will be used instead. Hence you have to look at the ports tab to see what the port is actually bound to 🔍
